aten/src/ATen/native/TensorConversions.cpp

```diff
@@ -144,6 +144,10 @@ Tensor view_dtype(const Tensor& self, ScalarType dtype) {
   const auto type_meta = c10::scalarTypeToTypeMeta(dtype);
   TORCH_CHECK(self.element_size() == static_cast<int64_t>(type_meta.itemsize()),
     "Viewing a tensor as a new dtype with a different number of bytes per element is not supported.");
+  TORCH_CHECK(!self.is_conj(),
+    "torch.Tensor.view is not supported for conjugate view tensors when converting to non-complex dtype.");
```

Comment:
You're not checking that this is converting to a non-complex dtype.
Is it ok under the assumption that there is only one complex dtype of each size?

Reply:
there are two possible cases:
1. complex_dtype1 -> complex_dtype1 (in which cases it returns from the first if)
2. complex_dtype1 -> complex_dtype2 (which would error out from the TORCH_CHECK above)

Reply:
offline sync: Disallow conversion of a conjugate view tensor to a different dtype since we are interpreting memory to a different format and it doesn't make sense to do that when the true value is not saved in memory as is the case for conjugate view tensors.